projects/components/src/form/number-with-unit-input/number-with-unit-form-input.component.ts

```diff
@@ -239,6 +239,10 @@ export class NumberWithUnitFormInputComponent extends BaseFormControl implements
         this.unlimitedControlValue = unlimitedChecked;
         this.textInputValue = unlimitedChecked ? '' : this.lastRealValue?.toString() || '';
 
+        if (!this.unitsControlValue && !unlimitedChecked) {
```

Comment:
Some comment will help understand why in this case we should set the value by getting the value